### PR TITLE
feat: drive reward claim and settlement process

### PR DESCRIPTION
**Pull request by djordon-stacks**
## Description

Closes https://github.com/stx-labs/spox/issues/43

## Changes

* Add Rust code for "driving" the reward claiming pipeline.
* Moved the existing deposit monitor stuff, with small modifications, to go through the new code.

## Testing Information

After the modifications, I built and ran spox, and things worked as expected. Specifically, I did this (taken from the README.md):
1. Built spox.
2. Ran sBTC devenv by running `make devenv-up` at the root of the sbtc repository.
3. Waited until Epoch 3.0 came. I checked the Bitcoin mempool explorer at http://localhost:8083/, but I could've done: 
    ```
    curl -s http://localhost:20443/v2/pox | jq ".current_epoch"
    ```
5. Got the signers' current aggregate key with
    ```
    ./target/debug/spox -c src/config/default.toml get-signers-xonly-key
    ```
7. Took that and replaced the current value for `signers_xonly` in `src/config/default.toml#L66` https://github.com/stx-labs/spox/blob/abe88f7ae6b9d0ba0b1d1821820c7456881c5a59/src/config/default.toml#L66
8. Got the deposit address with
    ```
    ./target/debug/spox -c src/config/default.toml get-deposit-address --network regtest
    ```
10. Ran spox with
    ```
    RUST_LOG=spox=trace,debug cargo run -- -c src/config/default.toml
    ```
12. Broadcast a deposit address using the sBTC demo CLI. It just sends BTC to some address. For me, the command was the following, but your recipient address should be different:
    ```
    # From the root of the sbtc repo
    ./target/debug/demo-cli fund-btc --recipient bcrt1pmt334klk5anfes5846k3s6jxgfm0hgury82m3vctaw2lwln7dh5qezvzk7
    ``` 

## Checklist

- [x] I have performed a self-review of my code
- [x] I have had an AI agent review my code

### CODE REVIEW

**Review by hstove-stacks — APPROVED** [review]
LGTM! The copilot comments seem like more edge-case type things, up to you if you want to handle them

File: src/dispatch.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+use crate::bitcoin::BlockRef;
+use crate::context::Context;
+
+/// Spawn a process that runs a function whenever a new Bitcoin chain
+/// tip is detected.
+pub async fn run_on_chain_tips<F, O>(func: F, mut rx: Receiver<BlockRef>, context: Context)
+where
+    F: Fn(mpsc::Receiver<BlockRef>, Context) -> O + Send + 'static,
+    O: Future + Send + 'static,
+    O::Output: Send + 'static,
```
[the remark this answers is not in the document]
Comment (djordon-stacks) [reply src/dispatch.rs:18]: It compiles

**Comment by djordon-stacks** [comment]
> LGTM! The copilot comments seem like more edge-case type things, up to you if you want to handle them

Na. At least not now. I want to add tests, but I'm a bit away from doing that.

File: src/bitcoin/chain_tip.rs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    let mut last_seen: Option<BlockRef> = None;
+
+    loop {
+        match rpc.get_chain_tip() {
+            Ok(tip) if Some(&tip) != last_seen.as_ref() => {
+                tracing::debug!(%tip, "detected new bitcoin chain tip");
+                match sender.send(tip) {
+                    Ok(_) => last_seen = Some(tip),
+                    Err(_) => tracing::warn!("no active subscribers for chain tip broadcast"),
+                }
```
[the remark this answers is not in the document]
Comment (djordon-stacks) [reply src/bitcoin/chain_tip.rs:59]: Oh, I'd do it differently, but yeah, this is not a bad suggestion either.

**Review by hstove-stacks — APPROVED** [review]
